metrics/matthews_correlation/matthews_correlation.py

```diff
@@ -97,7 +121,20 @@ def _info(self):
             ],
         )
 
-    def _compute(self, predictions, references, sample_weight=None):
-        return {
-            "matthews_correlation": float(matthews_corrcoef(references, predictions, sample_weight=sample_weight)),
-        }
+    def _compute(self, predictions, references, average=None, sample_weight=None):
+        if self.config_name == "multilabel":
+            references = np.array(references)
+            predictions = np.array(predictions)
+            if references.ndim != 2 or predictions.ndim != 2:
```

Comment:
I find this slighlty easier to read:
```suggestion
            if not (references.ndim == 2 and predictions.ndim == 2):
```